reactioncommerce/reaction
# Fix 3854: password validation

Resolves #3854  
Impact: **major**  
Type: **bugfix**

## Issue
In password validation, the presence of password was checked like `password.length > 0`. This was causing problems when the password was `undefined`.

## Solution
Changed the check to `password && password.length > 0`.

## Breaking changes
None

## Testing
1. Do the following as anonymous user.
1. Add item to cart
2. Goto checkout
3. Add email to register UI.  Do not add password, click "register".
4. Observe no error in the console.
5. Observe the validation message that "Password is required"
